earth_enterprise/src/fusion/dbmanifest/dbmanifest.cpp

```diff
@@ -327,6 +330,51 @@ void DbManifest::GetManifest(geFilePool &file_pool,
   GetXmlManifest(stream_manifest);
 }
 
+// Read POI file and extract the data file names from within it and then add
+// them to the manifest too
+void DbManifest::GetPoiDataFiles(const std::string& poi_file,
+                     std::vector<ManifestEntry>* search_manifest)
+{
+  notify(NFY_DEBUG,
+        "Parsing poi file %s looking for data files", poi_file.c_str());
+  khxml::DOMLSParser *parser = CreateDOMParser();
+  if (parser) {
+    khxml::DOMDocument *doc = ReadDocument(parser, poi_file);
+    if (doc) {
+      try {
+        if (khxml::DOMElement *root = doc->getDocumentElement()) {
+          if (khxml::DOMElement* el_search_table = GetFirstNamedChild(root, "SearchTableValues")) {
+            std::vector<std::string> data_files;
+            FromElementWithChildName(el_search_table, "SearchDataFile", data_files);
+            if (data_files.empty()) {
+              notify(NFY_WARN,
+                    "No data files defined for poi file %s", poi_file.c_str());
+            }
+            for (uint i = 0; i < data_files.size(); ++i) {
+              const std::string& data_file = data_files[i];
+              search_manifest->push_back(ManifestEntry(data_file));
+            }
+          } else {
+            notify(NFY_WARN,
+                  "No search table element found in poi file %s", poi_file.c_str());
+          }
+        } else {
+          notify(NFY_WARN,
+                 "No document element loading poi file %s", poi_file.c_str());
+        }
+      } catch(const std::exception &e) {
+        notify(NFY_WARN, "%s while loading poi file %s", e.what(), poi_file.c_str());
+      } catch(...) {
+        notify(NFY_WARN, "Unable to load poi file %s", poi_file.c_str());
+      }
+    } else {
+      notify(NFY_WARN, "Unable to read poi file %s", poi_file.c_str());
+    }
+    DestroyParser(parser);
+  } else {
+    notify(NFY_WARN, "Unable to get parser for poi file %s", poi_file.c_str());
```

Comment:
Certainly it is possible, but not tragic, if we fail to read the poi_file into the parser (readDocument returns a nullptr).  It could be malformed, it could have wrong file perms, it could even be missing.  But if we actually fail to create a parser, that seems like something must be broken inside the app (corrupted memory, out of memory, etc).

Reply:
Agreed.  Returning NULL (`nullptr` is C++11 BTW ;) ) from `CreateDOMParser` means something really bad is going on outside of the scope of this function.  Maybe something more harsh than a warning level debug print is inline for this?  There is no `NFY_ERROR` so the next step up is `NFY_FATAL` and that aborts the process.  Could do that or could raise an exception.